### PR TITLE
Add SQ-blocking dashboard to testgrid

### PR DESCRIPTION
**Pull request by krzyzacy**
per discussion from #535 #1383

### CODE REVIEW

**Comment by krzyzacy** [comment]
seems `Bazel test` is running the same test as travis, do we still need travis?

**Comment by spxtr** [comment]
Which test is duplicated?

File: testgrid/config/config.yaml [hunk 1]
```diff
@@ -1986,6 +1986,29 @@ dashboards:
   - name: aws-janitor
     test_group_name: maintenance-ci-aws-janitor
 
+- name: submit queue blocking
```
Comment (fejta) [inline testgrid/config/config.yaml]: Please don't be so verbose and also don't add spaces. So how about `blocking` or `sq-blocking`?

**Comment by fejta** [comment]
Before we create this tab I would like there to be a unit test that validates that parses https://github.com/kubernetes/contrib/blob/master/mungegithub/submit-queue/deployment/kubernetes/configmap.yaml and ensures that all `[submit-queue.jenkins-jobs](https://github.com/kubernetes/contrib/blob/master/mungegithub/submit-queue/deployment/kubernetes/configmap.yaml#L152)` items are present

**Comment by spxtr** [comment]
> Before we create this tab I would like there to be a unit test that validates that parses [kubernetes/configmap.yaml](https://github.com/kubernetes/contrib/blob/master/mungegithub/submit-queue/deployment/kubernetes/configmap.yaml) and ensures that all [`submit-queue.jenkins-jobs`](https://github.com/kubernetes/contrib/blob/master/mungegithub/submit-queue/deployment/kubernetes/configmap.yaml#L152) items are present 

It's not super easy to do since they're in separate repos. How do we update them atomically?

**Comment by krzyzacy** [comment]
Against contrib master? And one in contrib against test-infra master?

Or add a ci job which will signal a failure?

**Comment by krzyzacy** [comment]
hummmmm... so for some reason I cannot use vendor imports in go_test code here? It works locally.

**Comment by krzyzacy** [comment]
Ok it passes now ( I had to create a wrapper function ), seems like in go_test, import from vendor is invalid? 

And I added some test against contrib master to validate sq-blocking dashboard.

**Comment by fejta** [comment]
Looks great! Consider adding a comment in contrib/ repo telling people who modify that file to update this file.

/lgtm

**Comment by spxtr** [comment]
I'm not a fan of that unit test. Now changes to contrib can and will break tests in test-infra.

**Comment by krzyzacy** [comment]
Em.. Move it to CI will be a waste of resource.. Any better ideas? How about I add a comment to `configmap.yaml` and I'll own this test.

**Comment by fejta** [comment]
I agree it is far less than ideal, but I don't see a good alternative. And given the small number of test-infra contributors the gains of ensuring that the blocking testgrid dashboard is current outweighs the annoyance of having the testgrid tests fail unexpectedly (they are easy to fix)

**Comment by fejta** [comment]
How is this coming along?

**Comment by krzyzacy** [comment]
Waiting for https://github.com/kubernetes/contrib/pull/2248? I can get this one in.

**Comment by fejta** [comment]
That one says it is waiting on this one

**Comment by krzyzacy** [comment]
Oh you unassigned yourself there.. I can merge this first and ask Antoine for lgtm in the other PR.

**Comment by fejta** [comment]
Let me know when this is ready for review

**Comment by krzyzacy** [comment]
PR rebased, same as before. PTAL

File: testgrid/config/config.yaml [hunk 2]
```diff
@@ -1949,6 +1949,31 @@ dashboards:
   - name: testgrid-config-upload
     test_group_name: testgrid-config-upload
 
+- name: sq-blocking
```
Comment (fejta) [inline testgrid/config/config.yaml]: This list does not match http://submit-queue.k8s.io/#/e2e (missing kops-aws)

**Comment by fejta** [comment]
/lgtm

Please add the missing job and pass tests

File: testgrid/config/config.yaml (hunk 2 above)
Comment (krzyzacy) [inline testgrid/config/config.yaml]: whoops.. sorry did not double check

**Comment by krzyzacy** [comment]
@k8s-bot bazel test this

**Comment by fejta** [comment]
Looks like the code fails gofmt

**Comment by spxtr** [comment]
@k8s-bot bazel test this

**Comment by spxtr** [comment]
You should set `gofmt` to run whenever you save your code. If you're using vim then install vim-go, dunno how to do it for other editors.

**Comment by krzyzacy** [comment]
yeah I'll fix it...

**Comment by fejta** [comment]
FYI please pushing the fix before leaving the note so that it doesn't go back into my PR dashboard TODO list :smile:

File: testgrid/config/config.yaml [hunk 3]
```diff
@@ -1949,6 +1949,33 @@ dashboards:
...
+  dashboard_tab:
+  - name: aws
```
Comment (fejta) [inline testgrid/config/config.yaml]: I would keep the kops part

Comment (krzyzacy) [inline testgrid/config/config.yaml]: done.

**Pull request merged**